### ISSUE TITLE
Getting `Module not found: Can't resolve 'fs'` error

### ISSUE THREAD

**Issue by lamberski** [cypress-visual-regression/cypress-visual-regression#156]
I'm using Cypress 10.x inside a Next.js 13.x project with TypeScript.

I getting an error `Module not found: Can't resolve 'fs'` when trying to run the tests. I tried using plugin version 2.0.x and Cypress 11.x and 12.x as well - same result. I tried installing 'fs' inside the project (`npm install --save-dev fs`) but that does not solve the issue either.

The plugin worked without any issues in version 1.7.x with Cypress 9.

Is it only me or is this issue a common occurrence with version 2.1.x?

```bash
> npx cypress run --component --browser electron

warn  - You have enabled experimental feature (scrollRestoration) in next.config.js.
warn  - Experimental features are not covered by semver, and may cause unexpected or broken application behavior. Use at your own risk.

<i> [webpack-dev-server] Project is running at:
<i> [webpack-dev-server] Loopback: http://127.0.0.1:8080/
<i> [webpack-dev-server] Content not from webpack is served from '/Users/user/Projects/<project-name>/public' directory

====================================================================================================

  (Run Starting)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:        12.3.0                                                                         │
  │ Browser:        Electron 106 (headless)                                                        │
  │ Node Version:   v16.19.0 (/opt/homebrew/Cellar/node@16/16.19.0/bin/node)                       │
  │ Specs:          1 found (Button.test.tsx)                                                      │
  │ Searched:       **/*.test.{ts,tsx}                                                             │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                    
  Running:  Button.test.tsx                                                                 (1 of 1)
<i> [webpack-dev-middleware] wait until bundle finished: /__cypress/src/index.html
45 assets
696 modules

ERROR in ./node_modules/cypress-visual-regression/dist/utils.js:1:0
Module not found: Can't resolve 'fs'

Import trace for requested module:
./node_modules/cypress-visual-regression/dist/command.js
./cypress/component/support.ts
../../../Library/Caches/Cypress/12.3.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/@cypress/webpack-dev-server/dist/loader.js!../../../Library/Caches/Cypress/12.3.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/@cypress/webpack-dev-server/dist/browser.js
../../../Library/Caches/Cypress/12.3.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/@cypress/webpack-dev-server/dist/browser.js


https://nextjs.org/docs/messages/module-not-found

client (webpack 5.74.0) compiled with 1 error in 3105 ms

**Comment by MkMan** [comment]
I also have the same issue using the library for component tests, it works fine for e2e tests though.

The problem is 

```ts
import compareSnapshotCommand from 'cypress-visual-regression/dist/command'; // <- imports utils that import modules not present in the browser

compareSnapshotCommand();
```

I tried both with Webpack and Vite and has the same result.

**Comment by malykhin** [comment]
I faced the same issue with webpack 5. Resolved by adding: [NodePolyfillPlugin](https://github.com/Richienb/node-polyfill-webpack-plugin#readme) to plugins section of the webpack config along with:
```
resolve: {
    ...
    fallback: {
      fs: false
    }
}
```

**Comment by lamberski** [comment]
@malykhin Thank you for pointing us in the right direction!

It's indeed a thing related to Webpack 5 which does not come with node polyfills out-of-the-box.

In my case, all I had to do in Next.js is below:

```javascript
module.exports = {
    ...
    webpack(config) {
        config.resolve.fallback['fs'] = false
        return config
    },
}
```

In my experience, snapshots work with that one change. I didn't need to install the [node-polyfill-webpack-plugin](https://www.npmjs.com/package/node-polyfill-webpack-plugin).

**Comment by raimund-schluessler** [comment]
> It's indeed a thing related to Webpack 5 which does not come with node polyfills included out-of-the-box.

This does not really explain why cypress-visual-regression 1.7.0 works with the same webpack config, but also requires `fs` in `src/utils.js`.

**Comment by raimund-schluessler** [comment]
> Edit: With #139 `src/utils.js` is now newly required in `src/command.js` for importing the function `getValueOrDefault`. Could be that this introduced an implicit dependency to `fs` which was not there with 1.7.0.

This is indeed the problem. Replacing
```js
const { getValueOrDefault } = require('./utils');
```
with
```js
const getValueOrDefault = (value, defaultValue) => value || defaultValue;
```
in `command.js` (i.e. not implicitly requiring `fs` in `command.js`) makes it work again. I will send a PR later.

**Comment by raimund-schluessler** [comment]
PR is in #161.

**Comment by HeneryHawk** [comment]
The solution provided by @raimund-schluessler works for me.
Can you please merge the PR so that no manual fix is needed. This prevents me from using this tool as it will not work in CI pipelines...

**Comment by Zaista** [comment]
Released in v2.1.1
